File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) 
    
    {
        int n = nums.size();
        
        k = k % n;

        if (k == 0) return; 
        reverse(nums.begin(),nums.end()-k);
        reverse(nums.begin()+n-k,nums.end());
        reverse(nums.begin(),nums.end());
    }
};

File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
        const n=nums.length;
        const temp = new Array(n);


        nums.forEach((val,i)=>{
            temp[(i+k)%n]=val;
        })
        
        for (let i = 0; i < n; i++) {
            nums[i] = temp[i];
        }
};